packages/@glimmer/integration-tests/test/modifiers-test.ts

```diff
@@ -82,16 +83,223 @@ class ModifierTests extends RenderTest {
   }
 
   @test
-  'do not work on component invocations'(assert: Assert) {
-    this.registerComponent('Glimmer', 'Foo', '<div ...attributes>Foo</div>');
-    this.registerModifier('bar', BaseModifier);
-    assert.throws(() => {
-      this.render('<Foo {{bar foo="foo"}} />');
-    }, 'Compile Error: Element modifiers are not allowed in components');
-
-    assert.throws(() => {
-      this.render('<Foo (bar foo="foo") />');
-    }, 'Compile Error: Element modifiers are not allowed in components');
+  'modifiers on components are forwarded to a single element receiving the splattributes'(
+    assert: Assert
+  ) {
+    let modifierParams: Option<unknown[]> = null;
+    let modifierNamedArgs: Option<Dict<unknown>> = null;
+    let modifiedElement: SimpleElement | undefined;
+    class Bar extends AbstractInsertable {
+      didInsertElement(params: unknown[], namedArgs: Dict<unknown>) {
+        modifierParams = params;
+        modifierNamedArgs = namedArgs;
+        modifiedElement = this.element;
+      }
+    }
+    this.registerComponent('Glimmer', 'TheFoo', '<div id="inner-div" ...attributes>Foo</div>');
+    this.registerModifier('bar', Bar);
+    this.render('<TheFoo {{bar "something" foo="else"}}/>');
+    assert.deepEqual(modifierParams, ['something']);
+    assert.deepEqual(modifierNamedArgs, { foo: 'else' });
+    assert.equal(
+      modifiedElement && modifiedElement.getAttribute('id'),
+      'inner-div',
+      'Modifier is called on the element receiving the splattributes'
+    );
+  }
+
+  @test
+  'modifiers on components are forwarded to all the elements receiving the splattributes'(
+    assert: Assert
+  ) {
+    let elementIds: any[] = [];
```

Comment:
This can now be `string[]` (or maybe `Option<string>[]` – `getAttribute` returns `Option<string>`, which we can either just push into the array like you are doing now, or check for null first)